src/encode_task_preseq.py

```diff
@@ -0,0 +1,145 @@
+    parser.add_argument('--bam', type=str, help='Raw BAM file.')    
+    parser.add_argument('--out-dir', default='', type=str, help='Output directory.')
+    parser.add_argument('--log-level', default='INFO', help='Log level',
+                        choices=['NOTSET','DEBUG','INFO','WARNING','CRITICAL','ERROR','CRITICAL'])
+    args = parser.parse_args()
+    log.setLevel(args.log_level)
+    log.info(sys.argv)
+    return args
+
+def get_picard_complexity_metrics(aligned_bam, prefix):
+    '''
+    Picard EsimateLibraryComplexity
+    '''
+    # remove redundant (or malformed) info (read group) from bam
+    out_file = '{0}.picardcomplexity.qc'.format(prefix)
+    get_gc_metrics = ('mkdir -p tmp_java && java -Djava.io.tmpdir=$PWD/tmp_java -Xmx6G -XX:ParallelGCThreads=1 -jar '
+                      '{2} '
+                      'EstimateLibraryComplexity INPUT={0} OUTPUT={1} '
+                      'USE_JDK_DEFLATER=TRUE USE_JDK_INFLATER=TRUE '
+                      'VERBOSITY=ERROR '
+                      'QUIET=TRUE && rm -rf tmp_java').format(aligned_bam,
+                                           out_file,
+                                           locate_picard())
+    os.system(get_gc_metrics)
+
+    # Extract the actual estimated library size
+    header_seen = False
+    est_library_size = 0
+    with open(out_file, 'r') as fp:
+        for line in fp:
+            if header_seen:
+                est_library_size = int(float(line.strip().split()[-1]))
+                break
+            if 'ESTIMATED_LIBRARY_SIZE' in line:
+                header_seen = True
+
+    return est_library_size
+
+def run_preseq(bam_w_dups, prefix):
+    '''
+    Runs preseq. Look at preseq data output to get PBC/NRF.
+    '''
+    # First sort because this file no longer exists...
+    sort_bam = 'samtools sort -o {1}.sorted.bam -T {1} -@ 2 {0}'.format(
+        bam_w_dups, prefix)
+    os.system(sort_bam)
+
+    logging.info('Running preseq...')
+    preseq_data = '{0}.preseq.dat'.format(prefix)
+    preseq_log = '{0}.preseq.log'.format(prefix)
+    preseq = ('preseq lc_extrap '
+              '-P -B -o {0} {1}.sorted.bam -seed 1 -v 2> {2}').format(preseq_data,
+                                                              prefix,
+                                                              preseq_log)
+    logging.info(preseq)
+    os.system(preseq)
+    os.system('rm {0}.sorted.bam'.format(prefix))
+    return preseq_data, preseq_log
+
+def get_preseq_plot(data_file, prefix):
+    '''
+    Generate a preseq plot
+    '''
+    try:
+        data = np.loadtxt(data_file, skiprows=1)
+    except IOError:
+        return ''
+    data /= 1e6  # scale to millions of reads
+
+    fig = plt.figure()
+
+    # Plot the average expected yield
+    plt.plot(data[:, 0], data[:, 1], 'r-')
+
+    # Plot confidence intervals
+    ci_lower, = plt.plot(data[:, 0], data[:, 2], 'b--')
```

Comment:
I was a bit confused by this unpack because it is not obvious to me that `plt.plot(data[:, 0], data[:, 2], 'b--')` would always contain only one value. If this is written like this on purpose to raise the `ValueError` when `plt.plot(data[:, 0], data[:, 2], 'b--')` contains more than one value, checking explicitly and raising an exception would be more clear.